src/tribler/core/utilities/process_manager/manager.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+import logging
+import sqlite3
+import sys
+from pathlib import Path
+from threading import Lock
+from typing import ContextManager, List, Optional, Union
+
+from contextlib import contextmanager
+
+from tribler.core.utilities.process_manager import sql_scripts
+from tribler.core.utilities.process_manager.process import ProcessKind, TriblerProcess
+from tribler.core.utilities.process_manager.utils import with_retry
+
+logger = logging.getLogger(__name__)
+
+DB_FILENAME = 'processes.sqlite'
+
+
+global_process_manager: Optional[ProcessManager] = None
+
+_lock = Lock()
+
+
+def set_global_process_manager(process_manager: Optional[ProcessManager]):
+    global global_process_manager  # pylint: disable=global-statement
+    with _lock:
+        global_process_manager = process_manager
+
+
+def get_global_process_manager() -> Optional[ProcessManager]:
+    with _lock:
+        return global_process_manager
+
+
+def set_error(error: Union[str | Exception], replace: bool = False):
+    process_manager = get_global_process_manager()
+    if process_manager:
+        process_manager.current_process.set_error(error, replace)
+    else:
+        logger.warning('Cannot set error for process locker: no process locker global instance is set')
```

Comment:
NIT: could be replaced:

```suggestion
    if process_manager := get_global_process_manager():
        process_manager.current_process.set_error(error, replace)
    else:
        logger.warning('Cannot set error for process locker: no process locker global instance is set')
```

Or, even, by:

```python
def set_error(error: Union[str | Exception], replace: bool = False):
    if process_manager := get_global_process_manager():
        process_manager.current_process.set_error(error, replace)
        return

    logger.warning('Cannot set error for process locker: no process locker global instance is set')
```

Reply:
Done